src/main/scripts/install.sh

```diff
@@ -0,0 +1,215 @@
+#!/bin/sh
+
+#      Copyright (c) Microsoft Corporation.
+# 
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+# 
+#           http://www.apache.org/licenses/LICENSE-2.0
+# 
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+create_dmgr_profile() {
+    profileName=$1
+    hostName=$2
+    nodeName=$3
+    cellName=$4
+    adminUserName=$5
+    adminPassword=$6
+
+    ${WAS_ND_INSTALL_DIRECTORY}/bin/manageprofiles.sh -create -profileName ${profileName} -hostName $hostName \
+        -templatePath ${WAS_ND_INSTALL_DIRECTORY}/profileTemplates/management -serverType DEPLOYMENT_MANAGER \
+        -nodeName ${nodeName} -cellName ${cellName} -enableAdminSecurity true -adminUserName ${adminUserName} -adminPassword ${adminPassword}
+}
+
+add_admin_credentials_to_soap_client_props() {
+    profileName=$1
+    adminUserName=$2
+    adminPassword=$3
+    soapClientProps=${WAS_ND_INSTALL_DIRECTORY}/profiles/${profileName}/properties/soap.client.props
+
+    # Add admin credentials
+    sed -i "s/com.ibm.SOAP.securityEnabled=false/com.ibm.SOAP.securityEnabled=true/g" "$soapClientProps"
+    sed -i "s/com.ibm.SOAP.loginUserid=/com.ibm.SOAP.loginUserid=${adminUserName}/g" "$soapClientProps"
+    sed -i "s/com.ibm.SOAP.loginPassword=/com.ibm.SOAP.loginPassword=${adminPassword}/g" "$soapClientProps"
+
+    # Encrypt com.ibm.SOAP.loginPassword
+    ${WAS_ND_INSTALL_DIRECTORY}/profiles/${profileName}/bin/PropFilePasswordEncoder.sh "$soapClientProps" com.ibm.SOAP.loginPassword
+}
+
+create_systemd_service() {
```

Comment:
We should be able to use bin/wasservice.sh command instead of this function.

Reply:
What's the different? Is it a must to change? If yes, could you pls tell me the exact and equivalent command including all of the parameters?

Reply:
@majguo using WAS provided command will make it easier to maintain. Here is the command usage, you should be able to map current properties to this:
```
wasservice.sh -add service_name
   -serverName server_name
   -profilePath server_profile_directory
   [-wasHome app_server_root]
   [-startArgs additional_start_arguments]
   [-stopArgs additional_stop_arguments]
   [-userid user_id -password password]
```

Reply:
Thanks @git4rk , I will have a try today.

Reply:
@git4rk  I tested the above command (as well as reading the [command reference doc](https://www.ibm.com/docs/en/was-nd/9.0.5?topic=processes-wasservice-command)) but unfortunately failed. Here are the steps I took for testing:
* `ssh` into vm installed with the deployment manager
* sudo as root
* add service: `/datadrive/IBM/WebSphere/ND/V9/profiles/Dmgr001/bin/wasservice.sh -add dmgr -serverName dmgr -profilePath /datadrive/IBM/WebSphere/ND/V9/profiles/Dmgr001`
* start service: `systemctl start dmgr_was.init.service`
* get service status: `systemctl status dmgr_was.init.service`:
  ```
  dmgr_was.init.service - LSB: Starts and stops WebSphere Application Server instances
  Loaded: loaded (/etc/rc.d/init.d/dmgr_was.init; generated)
  Active: active (exited) since Mon 2021-05-10 07:49:46 UTC; 11s ago
  Docs: man:systemd-sysv-generator(8)
  Process: 6620 ExecStart=/etc/rc.d/init.d/dmgr_was.init start (code=exited, status=0/SUCCESS)

  May 10 07:49:46 dmgrf4c09bVM systemd[1]: Starting LSB: Starts and stops WebSphere Application Server instances... 
  May 10 07:49:46 dmgrf4c09bVM dmgr_was.init[6620]: Failure starting WebSphere Application Server - dmgr
  May 10 07:49:46 dmgrf4c09bVM dmgr_was.init[6620]: The service definition may be invalid - script /datadrive/IBM/WebSphere/ND/V9/profiles/Dmgr001/bin/startServer.sh
  May 10 07:49:46 dmgrf4c09bVM dmgr_was.init[6620]: could not be found or was not executable.
  May 10 07:49:46 dmgrf4c09bVM systemd[1]: Started LSB: Starts and stops WebSphere Application Server instances.
  ```

However, if I manually run the commands in script file `/etc/rc.d/init.d/dmgr_was.init`, I can start the deployment manager successfully. I don't know the why.

As a result, as the key point of the comment is to improve the maintenance, I would suggest putting it aside to move on our MVP release and then get back to it as a refactoring task in the future. What do you think? Cc @edburns .

Reply:
Created issue https://github.com/WASdev/azure.websphere-traditional.cluster/issues/12 to track this.